src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncIT.java

```diff
@@ -582,4 +589,48 @@ public void sync_withSameDraftsWithChangesInBatch_ShouldRetryUpdateBecauseOfConc
         assertThat(errorCallBackMessages).isEmpty();
         assertThat(warningCallBackMessages).isEmpty();
     }
+
+    @Test
+    public void sync_withProductBundle_shouldCreateProductReferencingExistingProduct() {
+        final ProductDraft productDraft = createProductDraftBuilder(PRODUCT_KEY_2_RESOURCE_PATH,
+            ProductType.referenceOfId(productType.getKey()))
+            .taxCategory(null)
+            .state(null)
+            .build();
+
+        // Creating the product reference attribute value
+        final ObjectNode productReference = JsonNodeFactory.instance.objectNode();
+        productReference.put(REFERENCE_TYPE_ID_FIELD, Product.referenceTypeId());
+        productReference.put(REFERENCE_ID_FIELD, product.getKey());
+
+        // Creating the attribute draft with the product reference
+        final String productReferenceAttributeName = "product-reference";
+        final AttributeDraft productReferenceAttribute =
+            AttributeDraft.of(productReferenceAttributeName, productReference);
+
+        // Creating the product variant draft with the product reference attribute
+        final ProductVariantDraft draftMasterVariant = productDraft.getMasterVariant();
+        assertThat(draftMasterVariant).isNotNull();
+        final List<AttributeDraft> attributes = draftMasterVariant.getAttributes();
+        attributes.add(productReferenceAttribute);
+        final ProductVariantDraft masterVariant = ProductVariantDraftBuilder.of(draftMasterVariant)
+                                                                            .attributes(attributes)
+                                                                            .build();
+
+        final ProductDraft productDraftWithProductReference = ProductDraftBuilder.of(productDraft)
+                                                                                 .masterVariant(masterVariant)
+                                                                                 .build();
+
+
+        final ProductSync productSync = new ProductSync(syncOptions);
+        final ProductSyncStatistics syncStatistics =
+            executeBlocking(productSync.sync(singletonList(productDraftWithProductReference)));
+
+        assertThat(syncStatistics.getReportMessage())
+            .isEqualTo(format("Summary: %d products were processed in total (%d created, %d updated and %d products"
```

Comment:
i'm really confused that we are testing report strings so strictly. I'd say testing reports should be more flexible, e.g. testing number of exceptions/errors/warning is correct - it is a strict info, but the report message itself should be customer specific, e.g. every customer should be able to format the report message as they want, using info from exception/error/warning lists.

Reply:
This report message is one defined by the sync library. This doesn't stop the user of the library from customizing their own message with the counters when they want. But I agree that it makes it easier for testing to assert on counters rather on the entire message string which could change later: 1bde451 For the rest of the assertions I made an issue for it to be done separatley: https://github.com/commercetools/commercetools-sync-java/issues/186